src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,12 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
After thinking a little bit more, I am not quite comfortable about this change Out Variable Declarations and Deconstruction features rely on BuildArgumentsForErrorRecovery for proper error recovery and, unlike lambdas,  they cannot rely on prior binding for this. So, I think we should enter this if even for generic method unless it is not constructed. Also, consider using IsGeneric helper rather than checking arity explicitly.